objectRecognitionSource/build.gradle

```diff
@@ -305,3 +305,38 @@ task copyTestResources(type: Copy, dependsOn: [configurations.testRuntime, confi
 }
 
 test.dependsOn copyModels, copyTestResources
+
+// Using gradle-dcompose-plugin to build objectRecognitionSource docker image
+dcompose {
+    if (rootProject.hasProperty('dockerRegistry')) {
+        def registryName = rootProject.getProperty('dockerRegistry')
+        objectRecognitionSource {
+            buildFiles = project.copySpec {
+                from "$projectDir/src/main/docker/"
+                from "$projectDir/build/distributions/objectRecognitionSource.tar"
+                if (rootProject.hasProperty('objectRecognitionModelFiles')) {
+                    def modelLocation = rootProject.getProperty('objectRecognitionModelFiles')
+                    from "$modelLocation"
+                } else {
+                    from "$projectDir/build/models"
+                }
```

Comment:
Here, you can either keep his one separate OR you can define a closure that does the model stuff. (`sourceSpecificAdditiions`} and if it's present, invoke it.  Then other things can do stuff similarly if necessary.

The generic one is invoked if present locally or something (that is `if project.hasProperty` kind-a thing